src/amd/compiler/aco_ir.h

```diff
@@ -856,6 +856,68 @@ enum block_kind {
    block_kind_needs_lowering = 1 << 12,
 };
 
+
+struct RegisterDemand {
+   constexpr RegisterDemand() = default;
+   constexpr RegisterDemand(const int16_t v, const int16_t s)
+      : vgpr{v}, sgpr{s} {}
+   int16_t vgpr = 0;
+   int16_t sgpr = 0;
+
+   constexpr friend bool operator==(const RegisterDemand a, const RegisterDemand b) {
+      return a.vgpr == b.vgpr && a.sgpr == b.sgpr;
+   }
+
+   RegisterDemand operator+(const Temp t) {
+      if (t.type() == vgpr)
```

Comment:
Maybe we should have made RegType type-safe.
The problem is that you named the vgpr RegDemand exactly the same as RegType::vgpr, and the stupid compiler picks the wrong one ;)
Please change this (and the ones below) to
`if (t.type() == RegType::sgpr)`

It should check for sgpr instead of vgpr, because we also have the RegClass::linear_vgpr which increases vgpr-demand. so, it's easier to only check for sgpr.

Reply:
Yikes, I will change that over the weekend ...

I need to get my test setup running